### ISSUE TITLE
needless_return_with_question_mark FN

### ISSUE THREAD

**Issue by samueltardieu** [rust-lang/rust-clippy#16958]
### Description

The following code should trigger `needless_return_with_question_mark`. Noticed while reviewing #16952.

```rust
fn expr_after_stmt() -> Result<(), ()> {
    return Err(())?;
    //~^ needless_return_with_question_mark
    Ok(())
}
```

@rustbot label +I-false-negative +C-bug
@rustbot claim

### Version

```text

```
